src/parser/SparqlLexer.h

```diff
@@ -63,13 +63,13 @@ class SparqlLexer {
   [[nodiscard]] bool empty() const;
 
   bool accept(SparqlToken::Type type);
-  bool accept(const std::string& raw, bool match_case = true);
+  bool accept(std::string_view raw, bool match_case = true);
   // Accepts any token
   void accept();
 
   // Checks whether the next token matches but does not consume it
-  bool peek(SparqlToken::Type type);
-  bool peek(const std::string& raw, bool match_case = true);
+  [[nodiscard]] bool peek(SparqlToken::Type type) const;
+  [[nodiscard]] bool peek(std::string_view raw, bool match_case = true) const;
```

Comment:
EDIT:
The thing below doesn't work, because there are cases where `accept` is called for "there might be a dot, but it is not required", where it is perfectly fine to ignore the return value. Please still read my comment for the future.

I know that the clang-tidy in Clion wants all const member functions to be nodiscard and also often put it there just to make the code less yellow. But what we should also consider to become better programmers: Which of our non-const functions should also be [[nodiscard]].

(I am just starting this train of thought).
In this concrete example:
I think `accept` should also be [[nodiscard]] because if you ignore the return value you have no idea whether a token was consumed or not, and if you have manually called `peek` before, that call was unnecessary because you could have called `accept` directly. 

But this is such a minor thought that I will still merge this code without it, especially since it is going to be thrown out soon and `accept` is not part of this PR:)

Reply:
TLDR: There's nothing for you to change here:)